src/Ssh/Ssh/SshCommands/EnterAzVMCommand.cs

```diff
@@ -88,11 +88,12 @@ public override void ExecuteCmdlet()
             base.ExecuteCmdlet();
 
             ValidateParameters();
-            SetResourceType();
- 
+            HandleTargetProperties();
```

Comment:
I think how I would structure it would be:
GenericResource targetResource = GetTargetResourceAndSetResourceType()
ConfigurePortNumberFromResourceTag(targetResource)

Here is my thought process: We have 3 tasks (get the target resource, set the resource type, set the port number). Setting the resource type and port number both rely on getting the resource, but it's hard to decouple getting the resource and setting the resource type, since we need some of the logic for setting the resource type in a scenario that there are more than one supported resource with the same name. So, I'm thinking that we can keep those two tasks in the same method and fully separate the logic to set the port.

What are your thoughts?